### PR TITLE
(FACT-2564) Add support for zpool_featureflags and fix zpool_version

### PR DESCRIPTION
**Pull request by smortex**
This allows facter-ng to return the same facts as facter 3.x.

This changes took place in facter 3.8.0 and was discussed here:
https://github.com/puppetlabs/facter/pull/1597

### CODE REVIEW

**Comment by oanatmaria** [comment]
Hi, could you please open a Jira ticket for this changes?

File: lib/resolvers/solaris/zpool_resolver.rb [hunk 1]
```diff
@@ -22,11 +22,13 @@ def zpool_fact(fact_name)
...
             output, _status = Open3.capture2('zpool upgrade -v')
             features_list = output.scan(/^\s+(\d+)/).flatten
+            features_flags = output.scan(/^([a-z0-9_]+)[[:blank:]]*(\(read-only compatible\))?$/).map(&:first)
 
             return if features_list.empty?
 
             @fact_list[:zpool_featurenumbers] = features_list.join(',')
-            @fact_list[:zpool_version] = features_list.last
+            @fact_list[:zpool_featureflags] = features_flags.join(',')
+            @fact_list[:zpool_version] = features_flags.any? ? 5000 : features_list.last
```
Comment (oanatmaria) [inline lib/resolvers/solaris/zpool_resolver.rb:31]: could you write some tests for these changes?

Comment (smortex) [reply lib/resolvers/solaris/zpool_resolver.rb]: Done!

**Comment by smortex** [comment]
I just opened https://tickets.puppetlabs.com/browse/FACT-2564, added tests as requested and rebased the whole on top of master with a reference to the Jira issue in the squashed commit message. Thanks!

File: spec/facter/resolvers/solaris/zpool_resolver_spec.rb [hunk 2]
```diff
@@ -28,6 +28,23 @@
...
+      let(:output) { load_fixture('zpool-with-featureflags').read }
+
+      it 'returns zpool version fact' do
+        result = Facter::Resolvers::Solaris::ZPool.resolve(:zpool_version)
+        expect(result).to eq('5000')
+      end
+
+      it 'returns zpool featureflags fact' do
+        result = Facter::Resolvers::Solaris::ZPool.resolve(:zpool_featureflags)
+        expect(result).to eq('async_destroy,empty_bpobj,lz4_compress,multi_vdev_crash_dump,' \
```
Comment (Filipovici-Andrei) [inline spec/facter/resolvers/solaris/zpool_resolver_spec.rb:42]: Could you move the expected feature flags into a separate let() like in the zpool_featureflags_spec?

Comment (smortex) [reply spec/facter/resolvers/solaris/zpool_resolver_spec.rb:42]: Sure!

**Review by oanatmaria — APPROVED** [review]

**Pull request merged**